### ISSUE TITLE
Database checks (and likely other flashes) persist from other instances of Bolt

### ISSUE THREAD

**Issue by Pinpickle** [bolt/bolt#3067]
I don't know how many instances of Bolt are running on my computer, but there's certainly more than one.

I have changed some of the contenttype.yml in one of them (instance A), but I haven't updated the database on it. I then go to a different instance (instance B) and get notified of a database update:
![image](https://cloud.githubusercontent.com/assets/3238878/6560543/d200af4a-c6ef-11e4-9955-614448c42bf6.png)

Clicking on the update link *then takes me to* instance A. In fact, instance B requires no db update at all:
![image](https://cloud.githubusercontent.com/assets/3238878/6560560/fc44e8e8-c6ef-11e4-93e8-22f7e50867ac.png)

So there's some sort of bleed between sessions, in this case. Probably likely only to happen in a dev environment as production side, things will be sandboxed and standalone. I've had slightly different occurances - such as going to localhost/instanceB/bolt (note the lack of trailing slash) will redirect me to localhost/instanceA/bolt/

**Comment by GawainLynch** [comment]
Can you also confirm it this got caught/fixed in #3073 / #3074?

**Comment by Pinpickle** [comment]
Unfortunately not! You get some interesting results (two differing database update messages from two different sites):

![image](https://cloud.githubusercontent.com/assets/3238878/6571724/5e5c3eca-c770-11e4-813a-44857726a4d6.png)

However, the flash that shouldn't be there *is* from an older version of Bolt so I'll have to do some more thorough testing on that one.

**Comment by CarsonF** [comment]
I think this has to do with the bolt session id's not being unique. I know I run into problems with authentications between my instances.

**Comment by bobdenotter** [comment]
These flashes are tied to your (symfony) session. So, i'm tempted to agree with Carsonf.

**Comment by CarsonF** [comment]
@bobdenotter Aren't cookies unique to the domain though? So that would only be the case for multiple instances on same domain. For example, `localhost:8000` and `localhost:8001`.

I guess we shouldn't assume that it's stored via a cookie. 

What about:

```php
sprintf('bolt_session_%s_%s', $_SERVER['SERVER_NAME'], $_SERVER['SERVER_PORT']);
```

**Comment by bobdenotter** [comment]
> Aren't cookies unique to the domain though? So that would only be the case for multiple instances on same domain.

Correct. 

> sprintf('bolt_session_%s_%s', $_SERVER['SERVER_NAME'], $_SERVER['SERVER_PORT']);

That's one option.. It still doesn't account for `example.org/a/bolt/` and `example.org/b/bolt`. I have a feeling this should be solved like we handle the options in `getAntiCSRFToken()` in Users.php.

**Comment by CarsonF** [comment]
> That's one option.. It still doesn't account for example.org/a/bolt/ and example.org/b/bolt, right?

Ah yeah you're right.

> I have a feeling this should be solved like we handle the options in getAntiCSRFToken() in Users.php.

Yeah that seems like what we need to do.

**Comment by cdowdy** [comment]
I just ran into this too running a ```dev.site.com``` and a ```site.com```, updated to 2.1.4 and templates/database everything was the same on both sites. the ```1.6.*``` versions didn't seem to have this problem or at least I didn't notice them

**Comment by GawainLynch** [comment]
Found the issue, and it will be addressed in the upcoming work on Bolt 2.3.

Tracking this via #2948

**Comment by bobdenotter** [comment]
Fixed by #3564
